### ISSUE TITLE
InvalidArgumentError (see above for traceback): Incompatible shapes: [16,160,160,64] vs. [16,159,159,64]

### ISSUE THREAD

**Issue by cclough** [qubvel/segmentation_models#19]
Thanks for making this great library. I'm trying to insert your Unet model into my own tensorflow training apparatus (generator, optimizer, loss function etc)...also within a Python 2.7 project (!)...and I get the following error: 

```
InvalidArgumentError (see above for traceback): Incompatible shapes: [16,160,160,64] vs. [16,159,159,64]
	 [[Node: optimization_1/gradients/outputs/link-resnet18/add_12/add_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _device="/job:localhost/replica:0/task:0/device:CPU:0"](optimization_1/gradients/outputs/link-resnet18/add_12/add_grad/Shape, optimization_1/gradients/outputs/link-resnet18/zero_padding2d_2/Pad_grad/Shape)]]
```
My input image size is 256x256. I've tried changing this to 200x200 and get similiar off-by-one error.

Any idea why this might be happening? Is there a specific input image size I must use?

**Comment by qubvel** [comment]
Hi, please provide tf and keras versions. and some code snippet will be very helpful to understand your problem.

**Comment by qubvel** [comment]
This is strange shape 160x160,  as I understand this is not the first block, so input have been downsanpled. But 160*4(two times downsampled)=640 and it is not 256?

**Comment by cclough** [comment]
Thanks Pavel.  Yes that was 256. If I change input chip size to 224, the error becomes:

`InvalidArgumentError (see above for traceback): All dimensions except 3 must match. Input 1 has shape [16 143 143 64] and doesn't match input 0 with shape [16 144 144 64].`

Then if I change input chip size to 200x200 it changes to:

`InvalidArgumentError (see above for traceback): All dimensions except 3 must match. Input 1 has shape [16 17 17 256] and doesn't match input 0 with shape [16 18 18 512].`

Im using
```
tensorflow                         1.8.0
Keras                              2.2.4
Keras-Applications                 1.0.6
Keras-Preprocessing                1.0.5
```

This is my code. I pass it as layer to tensorflow `sess.run([...])`
```

    from segmentation_models import Unet
    model = Unet(backbone_name='resnet18', encoder_weights='imagenet', classes=3, input_shape=(224,224,3))
    return tf.identity(model(X), name=output_name)

```

**Comment by qubvel** [comment]
`segmentation-models` not compatible with python 2.7, so i was not able to to test with such environment. But I was **not** able to reproduce your error with following:  

**Evironment**
```
Docker image: python:3.6
tensorflow==1.8.0 (cpu)
keras==2.2.4
scikit-image==0.14.1
segmentation-model==0.1.2
```
**Code**
```python
import numpy as np
from segmentation_models import Unet

x = np.ones((160, 224, 224, 3))
y = np.ones((160, 224, 224, 3))

model = Unet('resnet18', encoder_weights='imagenet',
                 classes=3,
                 input_shape=(224, 224, 3))

model.compile('Adam', 'binary_crossentropy', ['binary_accuracy'])
model.fit(x, y, epochs=3)
```

Everything works fine.

**Comment by cclough** [comment]
fixed it - not sure what was wrong before, but porting it all to the keras within tensorflow, and it works great e.g. `from tensorflow.python.keras.applications import vgg16`

**Comment by a-bowers** [comment]
I seem to have had a possibly similar problem today but resolved it . Just leaving this here in case it's useful to someone else like me who came across this issue. Using the following environment and the example code from the readme:
```
Python 3.7
Tensorflow 1.14.0 (cpu)
Keras 2.2.4
Segmentation models 0.2.1
```

My input images were 299x299, and I had the following:

`InvalidArgumentError: All dimensions except 3 must match. Input 1 has shape [16 19 19 256] and doesn't match input 0 with shape [16 20 20 512].`

Code above from 16 Oct 2018 worked fine for me too, so I assumed it must be a problem with my data... I tried resizing to 256x256 and got a more interesting error message more similar to cclough's initial one:

`InvalidArgumentError: Incompatible shapes: [16,256,256,1] vs. [16,256,256]`

This one gave me and idea, so I took a look at my arrays; my y data was of shape (n, 256, 256), so I added a new last dimension to change it to (n, 256, 256, 1), and it appears to work.
